x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/config_panel_header.tsx

```diff
@@ -0,0 +1,291 @@
+      frame.removeLayer(id);
+    }
+  });
+}
+
+function willLoseAllLayers(frame: FramePublicAPI, visualization: Visualization) {
+  return !getSuggestion(frame, visualization).layerId;
+}
+
+function getNewVisualizationState(frame: FramePublicAPI, visualization: Visualization) {
+  const { suggestion, layerId } = getSuggestion(frame, visualization);
+
+  dropUnusedLayers(frame, suggestion, layerId);
+
+  return visualization.initialize(frame, suggestion && suggestion.state);
+}
+
+function VisualizationSummary(props: Props) {
+  const visualization = props.visualizationMap[props.visualizationId || ''];
+
+  if (!visualization) {
+    return (
+      <>
+        {i18n.translate('xpack.lens.configPanel.chooseVisualization', {
+          defaultMessage: 'Choose a visualization',
+        })}
+      </>
+    );
+  }
+
+  return (
+    <NativeRenderer
+      render={visualization.renderDescription}
+      nativeProps={props.visualizationState}
+    />
+  );
+}
+
+function ConfirmationModal({
+  onCancel,
+  onConfirm,
+  selection,
+  visualizationMap,
+  frame,
+}: {
+  frame: FramePublicAPI;
+  selection: VisualizationSelection;
+  visualizationMap: Record<string, Visualization>;
+  onCancel: () => void;
+  onConfirm: () => void;
+}) {
+  const visualization = visualizationMap[selection.visualizationId];
+  const dropAll = willLoseAllLayers(frame, visualization);
+  const { label: visualizationLabel } = visualization.visualizationTypes.find(
+    v => v.id === selection.subVisualizationId
+  )!;
+
+  return (
+    <EuiOverlayMask>
+      <EuiConfirmModal
+        title={i18n.translate('xpack.lens.configPanel.dropLayersTitle', {
+          defaultMessage: 'Drop layers?',
+        })}
+        onCancel={onCancel}
+        onConfirm={onConfirm}
+        cancelButtonText={i18n.translate('xpack.lens.configPanel.dropLayersCancelLabel', {
+          defaultMessage: `No, keep layers`,
+        })}
+        confirmButtonText={i18n.translate('xpack.lens.configPanel.dropLayersConfirmLabel', {
+          defaultMessage: `Yes, drop layers`,
+        })}
+        defaultFocusedButton="cancel"
+        data-test-subj="lnsConfirmDropLayer"
+      >
+        <p>
+          {dropAll
+            ? i18n.translate('xpack.lens.configPanel.dropAllLayersDescription', {
+                defaultMessage: 'Switching to {visualizationLabel} will drop all layers.',
+                values: { visualizationLabel },
+              })
+            : i18n.translate('xpack.lens.configPanel.dropAllButLAstLayersDescription', {
+                defaultMessage:
+                  'Switching to {visualizationLabel} will drop all but the last layers.',
+                values: { visualizationLabel },
+              })}
+        </p>
+        <p>
+          {i18n.translate('xpack.lens.configPanel.dropLayersPrompt', {
+            defaultMessage: 'Are you sure you want to drop layers?',
+          })}
+        </p>
+      </EuiConfirmModal>
+    </EuiOverlayMask>
```

Comment:
Before improving the message here let's make sure the behavior is what we expect: this should show up when there is data loss generally